src/lib.rs

```diff
@@ -102,15 +102,23 @@ pub async fn console_print(buf: &[u8]) {
 }
 
 #[allow(clippy::explicit_write)]
-pub async fn send_image(write_sink: &mut Arc<Mutex<impl Write>>, buf: &[u8]) -> anyhow::Result<()> {
+pub async fn send_image(
+    write_sink: &mut Arc<Mutex<impl Write>>,
+    buf: &[u8],
+    quit: &Arc<Mutex<bool>>,
+) -> anyhow::Result<()> {
```

Comment:
Seeing two `Arc<Mutex<>>` vars it looks to me that there should be some kind of context instead.

Reply:
well, any of these two can be requested by any of the running threads..